inlanecruising_plugin/src/inlanecruising_plugin.cpp

```diff
@@ -428,6 +454,31 @@ std::vector<cav_msgs::TrajectoryPlanPoint> InLaneCruisingPlugin::trajectory_from
   return traj;
 }
 
+std::vector<PointSpeedPair> InLaneCruisingPlugin::attach_back_points(const std::vector<PointSpeedPair>& points, 
+                          const int nearest_pt_index, std::vector<inlanecruising_plugin::PointSpeedPair> future_points, double back_distance) const
+{
+  std::vector<PointSpeedPair> back_and_future;
+  back_and_future.reserve(points.size());
+  double total_dist = 0;
+  int min_i = 0;
+  for (int i = nearest_pt_index; i > 0; --i) { 
```

Comment:
Should this be >= ?

Reply:
I was wondering why is it that the case? wouldn't the idx try to access points[i-1] with -1 idx?

Reply:
fixed to do the same thing with fewer lines